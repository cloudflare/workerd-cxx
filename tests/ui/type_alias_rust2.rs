pub mod other_module {
    pub struct Source {
        member: u32,
    }
}

#[cxx::bridge]
mod ffi {
    extern "Rust" {
        // Not allowed - the target is not `extern "Rust"`.
        type Source = crate::other_module::Source;
    }
}

fn main() {}
