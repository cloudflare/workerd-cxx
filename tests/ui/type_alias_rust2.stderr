error[E0277]: the trait bound `other_module::Source: RustType` is not satisfied
  --> tests/ui/type_alias_rust2.rs:11:23
   |
11 |         type Source = crate::other_module::Source;
   |                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `RustType` is not implemented for `other_module::Source`
   |
note: required by a bound in `verify_rust_type`
  --> src/rust_type.rs
   |
   | pub fn verify_rust_type<T: RustType>() {}
   |                            ^^^^^^^^ required by this bound in `verify_rust_type`
